components/movementsensor/imuwit/imu.go

```diff
@@ -303,14 +328,6 @@ func (imu *wit) Close() error {
 	imu.logger.Debug("Closing wit motion imu")
 	imu.cancelFunc()
 	imu.activeBackgroundWorkers.Wait()
-
-	if imu.port != nil {
-		if err := imu.port.Close(); err != nil {
```

Comment:
The defer close logic lives in the start loop, so I moved that logic near there since it was doing the same thing twice. The logic needs to exist in one place and I opted for removing it from close rather than changing more of the structure of the update loop.

Reply:
Do you think I should move all that logic here?

Reply:
:shrug: No strong opinions. Feel free to move it wherever you want.